### ISSUE TITLE
Step 5 in Load Balancing Section

### ISSUE THREAD

**Issue by dumlutimuralp** [aws/aws-eks-best-practices#404]
**Describe the problem**
Not sure what exactly is meant by the below note in Load Balancing section. It needs clarification.

> When using an Ingress resource and Ingress Controller (like the AWS Load Balancer Controller) step 5 is handled by the relevant controller instead of kube-proxy

kube-proxy is the owner of iptables configuration on the node(s). And even with when the Pod IP is used as the target kube-proxy will still keep updating the iptables rules on the node(s) since a service is always based on ClusterIP. (NodePort is based on ClusterIP, LoadBalancer is based on NodePort and also Ingress is using the Service in the backend).

**References**
[Please include a link to the lines where the error appears.](https://github.com/aws/aws-eks-best-practices/blob/master/content/networking/loadbalancing/loadbalancing.md#pod-readiness)

**Comment by jicowan** [comment]
I don't think we're saying that iptables doesn't get updated when you use target type of IP instead of instance. Rather, the controller is registering the endpoint as an IP target with the load balancer and the load balancer is responsible for routing the traffic to the pod, not kube-proxy.

**Comment by dumlutimuralp** [comment]
> the controller is registering the endpoint as an IP target with the load balancer and the load balancer is responsible for routing the traffic to the pod, not kube-proxy.

Yes but this is not what is explained in Step 5.  Step 5 talks about how kube-proxy updates the iptables based on the changes to the endpoints. Then what the controller does in terms of register/deregister targets to the load balancer is a separate thing, it does not replace the actions in step 5, it is additional.

My recommendation is to reword the note as "When using ingress resource and Ingress Controller the controller takes ownership of registering/deregistering the target to the load balancer.".

**Comment by jicowan** [comment]
@dumlutimuralp can you submit a PR for this change?

**Comment by dumlutimuralp** [comment]
@jicowan , I just submitted the [PR](https://github.com/aws/aws-eks-best-practices/pull/422).

**Comment by dumlutimuralp** [comment]
We updated the whole section in 2024. Closing this issue.
